### PR TITLE
introduce hasStyle value

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

**Comment by IgorMinar** [comment]
needs to be rebased and CI is failing. otherwise it looks reasonable. @juliemr can you take a look too please?

**Comment by pkozlowski-opensource** [comment]
I would love to have this matcher, would come handy in few tests I've written before (ex. ones for NgStyle and some integration tests)

**Comment by matsko** [comment]
Rebased on master.

File: modules/angular2/src/testing/matchers.ts [hunk 1]
```diff
@@ -105,6 +107,33 @@ _global.beforeEach(function() {
       }
     },
 
+    toHaveCssStyle: function() {
+      return {compare: buildError(false), negativeCompare: buildError(true)};
```
Comment (juliemr) [inline modules/angular2/src/testing/matchers.ts]: Style nitpick: We don't really need the custom `negativeCompare` here. Instead I'd do:

```js
    toHaveCssStyle: function() {
      return {compare: {
        return function(actual, styles) {
          var allPassed;
          if (isString(styles)) {
            allPassed = DOM.hasStyle(actual, styles);
          } else {
            allPassed = !StringMapWrapper.isEmpty(styles);
            StringMapWrapper.forEach(styles, (style, prop) => {
              allPassed = allPassed && DOM.hasStyle(actual, prop, style);
            });
          }

          return {
            pass: allPassed,
            get message() {
              var expectedValueStr = isString(styles) ? styles : JSON.stringify(styles);
              return `Expected ${actual.outerHTML} ${!allPassed ? 'not ' : ''} to contain the
                      CSS ${isString(styles) ? 'property' : 'styles'} "${expectedValueStr}"`;
            }
          };
        };
      }
    },

**Comment by juliemr** [comment]
Looks good to me except for one optional nitpick.

File: modules/angular2/src/testing/matchers.ts (hunk 1 above)
Comment (matsko) [inline modules/angular2/src/testing/matchers.ts]: This makes sense up to the point where we use `!allPassed` to turn on the "not " token within the error message. That doesn't look right.

Comment (juliemr) [inline modules/angular2/src/testing/matchers.ts]: It's the recommended way in the Jasmine docs though - if it didn't pass but we're throwing an error, we know the check was with a `.not.`

Comment (matsko) [inline modules/angular2/src/testing/matchers.ts]: True. OK. I've updated the changes and force pushed.

File: modules/angular2/src/testing/matchers.ts [hunk 2]
```diff
@@ -105,6 +107,31 @@ _global.beforeEach(function() {
...
+            StringMapWrapper.forEach(styles, (style, prop) => {
+              allPassed = allPassed && DOM.hasStyle(actual, prop, style);
+            });
+          }
+
+          return {
+            pass: allPassed,
+            get message() {
+              var expectedValueStr = isString(styles) ? styles : JSON.stringify(styles);
+              return `Expected ${actual.outerHTML} ${!allPassed ? 'not ' : ''} to contain the
```
Comment (gkalpak) [inline modules/angular2/src/testing/matchers.ts]: I think it should be the other way around: `!allPassed ? '' : 'not'`

File: modules/angular2/src/testing/matchers.ts (hunk 1 above)
Comment (IgorMinar) [inline modules/angular2/src/testing/matchers.ts]: CI still not happy

File: modules/angular2/src/testing/matchers.ts [hunk 3]
```diff
@@ -105,6 +107,31 @@ _global.beforeEach(function() {
...
+            StringMapWrapper.forEach(styles, (style, prop) => {
+              allPassed = allPassed && DOM.hasStyle(actual, prop, style);
+            });
+          }
+
+          return {
+            pass: allPassed,
+            get message() {
+              var expectedValueStr = isString(styles) ? styles : JSON.stringify(styles);
+              return `Expected ${actual.outerHTML} ${!allPassed ? ' ' : 'not '}to contain the
```
Comment (gkalpak) [inline modules/angular2/src/testing/matchers.ts]: Pretty munor, but `' '` should be `''` to avoid excess whitespace.

**Pull request closed without merging**